task7/task7.js

```diff
@@ -0,0 +1,535 @@
+class AdCollection{
+
+    constructor (adList){
+        if(adList.length != 0){
+            this._adList = adList.slice();
+        }else{
+            this._adList = [];
+        }
+    }
+
+    _filterByVendors(ads, vendors,skip, top){
+        return ads.filter(сurrentValue => vendors.includes(сurrentValue.vendor)).slice(skip,skip+top);
+    }
+
+    _filterByTags (ads, hashTags, skip, top){
+        return ads.filter(currentAd =>{
+            return hashTags.some(hashTag =>{ return currentAd.hashTags.includes(hashTag)})
+        });
+    }
+
+    getPage(skip = 0, top = 10, filterConfig = {}){
+        let adsToShow = [];
+        let keys = Object.keys(filterConfig);
+    
+        if(keys.length == 0){
+            adsToShow = this._adList.slice(skip,skip+top);
+        }else if (keys.length == 1){
+            if(keys[0] == 'hashTags'){
+                adsToShow = this._filterByTags(this._adList, filterConfig[keys[0]],skip, top);
+            } else if(keys[0] == 'vendors'){
+                adsToShow = this._filterByVendors(this._adList, filterConfig[keys[0]],skip, top);
+            }
+        }else if(keys.length == 2){
+            if(keys[0] == 'hashTags'){
+                adsToShow = this._filterByTags(this._adList, filterConfig[keys[0]],skip, top);
+                if (keys[1] == 'vendors'){
+                    adsToShow = this._filterByVendors(adsToShow,filterConfig[keys[1]],0,top);
+                }
+            }else if(keys[0] == 'vendors'){
+                adsToShow = this._filterByVendors(this._adList, filterConfig[keys[0]],skip, top);
+                if (keys[1] == 'hashTags'){
+                    adsToShow = this._filterByTags(adsToShow,filterConfig[keys[1]],0,top);
+                }
```

Comment:
The same problem as in previous PR: there's no need to check whether `vendors` is first or second key in object. Just use simple constructions:
```
if (filterConfig.vendors) {
  //...doFilter();
}
if (filterConfig.hashTags) {
  //...anotherFilter();
}

```